### ISSUE TITLE
Can't enable anymore (Firewall shows 0x6D9)

### ISSUE THREAD

**Issue by rudolphos** [ionuttbara/windows-defender-remover#127]
## *What* affects the bug ?

**Windows Defender Security App** and **Windows Defender Firewall with Advanced Security**

## *When* does this occur?

When launching **Windows Defender Firewall with Advanced Security**
![image](https://github.com/ionuttbara/windows-defender-remover/assets/4366810/be21699e-c051-4086-856f-5ef6d656f4dd)
Shows error **0x6D9**
> There was an error opening the Windows Firewall with Advanced Security snap-in, The Windows Firewall with Advanced Security snap-in failed to load, Restart the Windows Firewall service on the computer that you are managing, Error 0x6D9.

This service can't be started:
> Windows could not start the Windows Defender Firewall service on Local Computer.
> Error 1068: The dependency service or group failed to start.

![image](https://github.com/ionuttbara/windows-defender-remover/assets/4366810/65e34afd-c186-45db-bc45-fda8e24fc8ee)
> Windows Defender Firewall can't change some of your settings.
> Error code 0x8007042c

And can't open Windows defender
![image](https://github.com/ionuttbara/windows-defender-remover/assets/4366810/089ffa09-60aa-47b0-876b-dc0085f35f35)

## *In which* version of Windows does this issue happen ? Write version complete.

Windows 10 22H2

## *How* do we replicate the issue?

Previously use Y option in the script -  `Press (Y) for removing Defender and Security Components (old method, breaking Windows Updates/UWP in some version of Windows, removes files and unregisters classes)
`
As that was the only way to disable Defender..

Now after trying to enable it using R option - `Press (R) for enabling Defender and Security Components (restore actions where N is pressed)
`
It doesn't enable anymore anything. And it seems Defender is permanently removed, the only way to repair 0x6D9 error seems to be a clean install of Windows or an in-place upgrade.

I've tried `sfc/ scannow` and `DISM` component repair without any success.
I also used Tweaking.com tool to repair Windows Firewall, of course without any success.

## Expected behavior (if you have any solution write here.)

It shouldn't have disabled **Windows Defender Firewall with Advanced Security** permanently since many .msi installers depend on it to add firewall rules.

For example this .MSI log file shows it.

```
ExecFirewallExceptions:  Entering ExecFirewallExceptions in C:\WINDOWS\Installer\MSIC1B2.tmp, version 3.11.4516.0
ExecFirewallExceptions:  Installing firewall exception2 Cloudflare WARP (C:\Program Files\Cloudflare\Cloudflare WARP\warp-svc.exe)
ExecFirewallExceptions:  Error 0x800706d9: failed trying to find existing app
ExecFirewallExceptions:  Error 0x800706d9: failed to add/update application exception for name 'Cloudflare WARP', file 'C:\Program Files\Cloudflare\Cloudflare WARP\warp-svc.exe'
CustomAction WixExecFirewallExceptionsInstall returned actual error code 1603 (note this may not be 100% accurate if translation happened inside sandbox)
Action ended 3:56:22: InstallFinalize. Return value 3.
ExecFirewallExceptions:  Entering ExecFirewallExceptions in C:\WINDOWS\Installer\MSIC1E2.tmp, version 3.11.4516.0
ExecFirewallExceptions:  Uninstalling firewall exception2 Cloudflare WARP (C:\Program Files\Cloudflare\Cloudflare WARP\warp-svc.exe)
ExecFirewallExceptions:  Error 0x800706d9: failed to remove authorized app
ExecFirewallExceptions:  Error 0x800706d9: failed to remove application exception for name 'Cloudflare WARP', file 'C:\Program Files\Cloudflare\Cloudflare WARP\warp-svc.exe'
CustomAction WixRollbackFirewallExceptionsInstall returned actual error code 1603 but will be translated to success due to continue marking
Info 1903.Scheduling reboot operation: Deleting file C:\Program Files\Cloudflare\Cloudflare WARP\TBDC1F2.tmp. Must reboot to complete operation.
Action ended 3:56:22: INSTALL. Return value 3.
MSI (s) (C0:8C) [03:56:22:212]: Product: Cloudflare WARP -- Installation failed.

MSI (s) (C0:8C) [03:56:22:213]: Windows Installer installed the product. Product Name: Cloudflare WARP. Product Version: 24.1.7.1. Product Language: 1033. Manufacturer: Cloudflare, Inc.. Installation success or error status: 1603.
```

**Comment by ionuttbara** [comment]
noted. a fix is coming

**Comment by ionuttbara** [comment]
fixed in 12.6.9

**Comment by clearminded-cloud** [comment]
I get similar issue where I mistakingly deleted "HKLM:\SOFTWARE\Microsoft\PolicyManager" registry key on a test server, then firewall won't start.  The firewall is stuck in "starting" loop.  To fix, I had to export "HKLM:\SOFTWARE\Microsoft\PolicyManager\default" from a healthy system.   I copied from another Windows 2022 server.  Import that default key, then reboot.  Finally firewall starts again.  I'll then also make sure the permissions on the key is the same as on the healthy system.  Note, when checking permissions, there will be a unresolved permission, which is expected, as it's a compatibility SID.  In future, instead of deleting a key, I'll just rename a key, then rename it back if there's an issue.
